src/haxelib/api/Installer.hx

```diff
@@ -0,0 +1,852 @@
+
+		try
+			installFromDependencies(dependencies)
+		catch (e)
+			throw new InstallationException('Failed installing dependencies from $path:\n$e');
+	}
+
+	/**
+		Installs the libraries required to build the HXML file at `path`.
+
+		Throws an error when trying to install a library from the haxelib
+		server if the library has no versions or if the requested
+		version does not exist.
+
+		If `confirmHxmlInstall` is passed in, it will be called with information
+		about the libraries to be installed, and the installation only proceeds if
+		it returns `true`.
+	 **/
+	public function installFromHxml(path:String, ?confirmHxmlInstall:(libs:Array<{name:ProjectName, version:Version}>) -> Bool) {
+		final path = FileSystem.fullPath(path);
+		userInterface.log('Installing all libraries from $path:');
+		final libsToInstall = LibFlagData.fromHxml(path);
+
+		if (libsToInstall.empty())
+			return;
+
+		// Check the version numbers are all good
+		userInterface.log("Loading info about the required libraries");
+
+		final installData = getFilteredInstallData(libsToInstall);
+
+		final libVersions = [
+			for (library in installData)
+				{name:library.name, version:library.version}
+		];
+		// Abort if not confirmed
+		if (confirmHxmlInstall != null && !confirmHxmlInstall(libVersions))
+			return;
+
+		for (library in installData) {
+			try
+				installFromInstallData(library.name, library.installData)
+			catch (e) {
+				userInterface.log(e.toString());
+				continue;
+			}
+
+			final libraryName = switch library.installData {
+				case VcsInstall(version, vcsData): getName(library.name, version, vcsData.subDir);
+				case _: library.name;
+			}
+
+			if (library.isLatest || !scope.isLibraryInstalled(libraryName))
+				setVersionAndLog(libraryName, library.installData);
+
+			userInterface.log("Done");
+
+			handleDependenciesGeneral(libraryName, library.installData);
+		}
+	}
+
+	/** Installs the latest version of `library` from the haxelib server. **/
+	public function installLatestFromHaxelib(library:ProjectName) {
+		final info = Connection.getInfo(library);
+		// get correct name capitalization so that the logged output is correct
+		final library = ProjectName.ofString(info.name);
+
+		final versions = [for (v in info.versions) v.name];
+
+		if (versions.length == 0)
+			throw new InstallationException('The library $library has not yet released a version');
+
+		final version = getLatest(versions);
+		downloadAndInstall(library, version);
+
+		scope.setVersion(library, version);
+
+		userInterface.log('  Current version is now $version');
+		userInterface.log("Done");
+
+		handleDependencies(library, version);
+	}
+
+	/**
+		Installs `version` of `library` from the haxelib server.
+
+		If `forceSet` is set to true and the installer is running with
+		a global scope, the new version is always set as the current one.
+	 **/
+	public function installFromHaxelib(library:ProjectName, version:SemVer, forceSet:Bool = false) {
```

Comment:
This is also very similar to `installLatestFromHaxelib` and could be merged.

Reply:
Merged them.